### PR TITLE
[MLCompute] Add support for Xcode 13 beta 4.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/MLCompute/MLHelpers.cs [hunk 1]
```diff
@@ -235,4 +235,23 @@ public static string GetDebugDescription (this MLCComparisonOperation self)
...
 		}
 	}
+
+#if NET
+	[SupportedOSPlatform ("ios15.0")]
+	[SupportedOSPlatform ("tvos15.0")]
+	[SupportedOSPlatform ("macos12.0")]
+#else
+	[NoWatch]
+	[TV (15,0), Mac (12,0), iOS (15,0)]
```
Comment (spouliot) [inline src/MLCompute/MLHelpers.cs:245]: catalyst ?

File: src/mlcompute.cs [hunk 2]
```diff
@@ -104,6 +110,7 @@ enum MLCDeviceType /* int32_t */ {
 		Cpu = 0,
 		Gpu = 1,
 		Any = 2,
+		Ane  = 3, // Apple neural engine
```
Comment (spouliot) [inline src/mlcompute.cs:113]: [iOS (15,0), TV (15,0), Mac (12,0), MacCatalyst (15,0)]

File: src/mlcompute.cs [hunk 3]
```diff
@@ -538,11 +596,21 @@ interface MLCDevice : NSCopying {
 		[return: NullAllowed]
 		MLCDevice GetDevice (IMTLDevice[] gpus);
 
+		[TV (15,0), Mac (12,0), iOS (15,0)]
```
Comment (spouliot) [inline src/mlcompute.cs:599]: catalyst ?

File: src/mlcompute.cs [hunk 4]
```diff
@@ -673,6 +741,31 @@ interface MLCTensor : NSCopying {
...
+
+		[TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("tensorByDequantizingToType:scale:bias:axis:")]
+		[return: NullAllowed]
+		MLCTensor CreateByDequantizing (MLCDataType type, MLCTensor scale, MLCTensor bias, nint axis);
+
+		[Static]
+		[TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("tensorWithShape:randomInitializerType:dataType:")]
+		MLCTensor Create (NSNumber[] shape, MLCRandomInitializerType randomInitializerType, MLCDataType dataType);
```
Comment (spouliot) [inline src/mlcompute.cs:768]: `[BindAs (typeof (nint[]))] NSNumber[] shape` was used for `tensorWithShape:`

File: src/mlcompute.cs [hunk 5]
```diff
@@ -2282,4 +2375,48 @@ interface MLCSelectionLayer {
...
 		MLCSelectionLayer Create ();
 	}
+
+	[TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLCPlatform {
+		
+		[Static]
+		[Export ("setRNGSeedTo:")]
+		void SetRngSeedTo ([BindAs (typeof (nuint))] NSNumber seed);
```
Comment (spouliot) [inline src/mlcompute.cs:2385]: -> `SetRngSeed`

File: src/mlcompute.cs [hunk 6]
```diff
@@ -2282,4 +2375,48 @@ interface MLCSelectionLayer {
 		[Export ("layer")]
 		MLCSelectionLayer Create ();
 	}
+
+	[TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
```
Comment (spouliot) [inline src/mlcompute.cs:2380]: `[DisableDefaultCtor]` since there's no instance member defined

File: src/mlcompute.cs [hunk 7]
```diff
@@ -2282,4 +2375,48 @@ interface MLCSelectionLayer {
...
+		[Export ("setRNGSeedTo:")]
+		void SetRngSeedTo ([BindAs (typeof (nuint))] NSNumber seed);
+
+		[return: BindAs (typeof (nuint)), NullAllowed]
+		[Static]
+		[Export ("getRNGseed")]
+		NSNumber GetRngSeed ();
+	}
+
+	[TV (15,0), Mac (12,0), iOS (15,0)]
```
Comment (spouliot) [inline src/mlcompute.cs:2393]: catalyst ?

File: src/mlcompute.cs [hunk 8]
```diff
@@ -2282,4 +2375,48 @@ interface MLCSelectionLayer {
...
+		[Export ("timeStep")]
+		nuint TimeStep { get; }
+
+		[Static]
+		[Export ("optimizerWithDescriptor:")]
+		MLCAdamWOptimizer GetOptimizer (MLCOptimizerDescriptor optimizerDescriptor);
+
+		[Static]
+		[Export ("optimizerWithDescriptor:beta1:beta2:epsilon:usesAMSGrad:timeStep:")]
+		MLCAdamWOptimizer GetOptimizer (MLCOptimizerDescriptor optimizerDescriptor, float beta1, float beta2, float epsilon, bool usesAMSGrad, nuint timeStep);
```
Comment (spouliot) [inline src/mlcompute.cs:2419]: -> `usesAmsGrad`

File: tests/xtro-sharpie/common-MLCompute.ignore [hunk 9]
```diff
@@ -8,3 +8,5 @@
 !missing-enum-value! MLCPoolingType native value MLCPoolingTypeCount = 4 not bound
 !missing-enum-value! MLCRandomInitializerType native value MLCRandomInitializerTypeCount = 4 not bound
 !missing-enum-value! MLCReductionType native value MLCReductionTypeCount = 10 not bound
+!missing-enum-value! MLCDeviceType native value MLCDeviceTypeCount = 4 not bound
+!missing-enum-value! MLCDataType native value MLCDataTypeCount = 10 not bound
```
Comment (spouliot) [inline tests/xtro-sharpie/common-MLCompute.ignore:12]: remove the comments (a few lines earlier)
looks like they were disabled, likely in a beta bump

**Review by spouliot — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Unrelated issue: https://github.com/xamarin/maccore/issues/2443

**Pull request merged**